packages/cardano-services/src/ChainHistory/DbSyncChainHistoryProvider/ChainHistoryBuilder.ts

```diff
@@ -0,0 +1,187 @@
+    hashes: Cardano.TransactionId[]
+  ): Promise<TransactionDataMap<Cardano.Withdrawal[]>> {
+    const byteHashes = hashes.map((hash) => hexStringToBuffer(hash.toString()));
+    this.#logger.debug('About to find withdrawals for txs:', byteHashes);
+    const result: QueryResult<WithdrawalModel> = await this.#db.query(Queries.findWithdrawal, [byteHashes]);
+    const withdrawalMap: TransactionDataMap<Cardano.Withdrawal[]> = new Map();
+    for (const withdrawal of result.rows) {
+      const txId = Cardano.TransactionId(withdrawal.tx_id.toString('hex'));
+      const currentWithdrawals = withdrawalMap.get(txId) ?? [];
+      withdrawalMap.set(txId, [...currentWithdrawals, mapWithdrawal(withdrawal)]);
+    }
+    return withdrawalMap;
+  }
+
+  public async queryRedeemersByHashes(
+    hashes: Cardano.TransactionId[]
+  ): Promise<TransactionDataMap<Cardano.Redeemer[]>> {
+    const byteHashes = hashes.map((hash) => hexStringToBuffer(hash.toString()));
+    this.#logger.debug('About to find redeemers for txs:', byteHashes);
+    const result: QueryResult<RedeemerModel> = await this.#db.query(Queries.findRedeemer, [byteHashes]);
+    const redeemerMap: TransactionDataMap<Cardano.Redeemer[]> = new Map();
+    for (const redeemer of result.rows) {
+      const txId = Cardano.TransactionId(redeemer.tx_id.toString('hex'));
+      const currentRedeemers = redeemerMap.get(txId) ?? [];
+      redeemerMap.set(txId, [...currentRedeemers, mapRedeemer(redeemer)]);
+    }
+    return redeemerMap;
+  }
+
+  public async queryTxMetadataByHashes(
+    hashes: Cardano.TransactionId[]
+  ): Promise<TransactionDataMap<Cardano.TxMetadata>> {
+    const byteHashes = hashes.map((hash) => hexStringToBuffer(hash.toString()));
+    this.#logger.debug('About to find metadata for txs:', hashes);
+    const result: QueryResult<TxMetadataModel> = await this.#db.query(Queries.findTxMetadata, [byteHashes]);
+    if (result.rows.length === 0) return new Map();
+    const metadataMap: TransactionDataMap<TxMetadataModel[]> = new Map();
+    for (const metadata of result.rows) {
+      const txId = Cardano.TransactionId(metadata.tx_id.toString('hex'));
+      const currentMetadata: TxMetadataModel[] = metadataMap.get(txId) ?? [];
+      metadataMap.set(txId, [...currentMetadata, metadata]);
+    }
+    return new Map([...metadataMap].map(([id, metadata]) => [id, mapTxMetadata(metadata)]));
+  }
+
+  public async queryCertificatesByHashes(
+    hashes: Cardano.TransactionId[]
+  ): Promise<TransactionDataMap<Cardano.Certificate[]>> {
+    const byteHashes = hashes.map((hash) => hexStringToBuffer(hash.toString()));
+    this.#logger.debug('About to find certificates for txs:', byteHashes);
+    const poolRetireCerts: QueryResult<PoolRetireCertModel> = await this.#db.query(Queries.findPoolRetireCerts, [
+      byteHashes
+    ]);
+
+    const poolUpdateCerts: QueryResult<PoolUpdateCertModel> = await this.#db.query(Queries.findPoolUpdateCerts, [
```

Comment:
better to rename it to poolRegisterCerts

Reply:
Done in 70dd1e23cbb08257e051305e215dc03c001d0d99